website/server/models/task.js

```diff
@@ -69,6 +69,12 @@ export let TaskSchema = new Schema({
     broken: {type: String, enum: ['GROUP_DELETED', 'TASK_DELETED', 'UNSUBSCRIBED']},
     assignedUsers: [{type: String, ref: 'User', validate: [validator.isUUID, 'Invalid uuid.']}],
     taskId: {type: String, ref: 'Task', validate: [validator.isUUID, 'Invalid uuid.']},
+    requiresApproval: {type: Boolean, default: false},
+    approved: {type: Boolean, default: false},
+    approvedDate: {type: Date},
+    approvingUser: {type: String, ref: 'User', validate: [validator.isUUID, 'Invalid uuid.']},
+    approvalRequested: {type: Boolean, default: false},
+    approvalRequestedDate: {type: Date},
```

Comment:
Why no flexibility? For subdivision I meant an object not a separate model in a separate file